### PR TITLE
isort timeseries

### PR DESCRIPTION
**Pull request by nstarman**
Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
There is a conflict. FYI.

File: astropy/timeseries/periodograms/base.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import abc
+
 import numpy as np
-from astropy.timeseries import TimeSeries, BinnedTimeSeries
+
+from astropy.timeseries.binned import BinnedTimeSeries
```
Comment (pllim) [inline astropy/timeseries/periodograms/base.py:5]: Huh, why would isort complain about this one?

File: astropy/timeseries/binned.py [hunk 2]
```diff
@@ -4,12 +4,11 @@
 
 import numpy as np
 
-from astropy.table import groups, Table, QTable
-from astropy.time import Time, TimeDelta
 from astropy import units as u
-from astropy.units import Quantity
-
+from astropy.table import QTable, Table, groups
```
Comment (pllim) [inline astropy/timeseries/binned.py:8]: Doesn't "g" come before "Q"?

Comment (nstarman) [reply astropy/timeseries/binned.py:8]: Classes come first

Comment (pllim) [reply astropy/timeseries/binned.py:8]: Ooops. I think I did the opposite of that in https://github.com/astropy/astropy/pull/13764 and https://github.com/astropy/astropy/pull/13765 but CI didn't complain, so maybe it's ok.

File: astropy/timeseries/periodograms/base.py (hunk 1 above)
Comment (nstarman) [reply astropy/timeseries/periodograms/base.py:5]: `TimeSeries` wasn't used and astropy is first-party, so it's separated from third-party.

Comment (pllim) [reply astropy/timeseries/periodograms/base.py:5]: No, I mean why it made you go from `from astropy.timeseries import BinnedTimeSeries` to `from astropy.timeseries.binned import BinnedTimeSeries`?

Comment (nstarman) [reply astropy/timeseries/periodograms/base.py:5]: Oh. That was a huge pain. time series has an import cycle that's preventable by importing directly from the source.

**Review by pllim — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**